File: libs/typeorm/src/entities/ChannelEmoteUsage.ts
/* eslint-disable import/no-cycle */
import { Column, CreateDateColumn, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';

import { Channel } from './Channel';
import { User } from './User';

@Entity('channels_emotes_usages', { schema: 'public' })
export class ChannelEmoteUsage {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => Channel, _ => _.emotesUsages, {
    onDelete: 'RESTRICT',
    onUpdate: 'CASCADE',
  })
  @JoinColumn([{ name: 'channelId' }])
  channel?: Channel;

  @Column()
  channelId?: string;

  @ManyToOne(() => User, _ => _.emotesUsages, {
    onDelete: 'RESTRICT',
    onUpdate: 'CASCADE',
  })
  @JoinColumn([{ name: 'userId' }])
  user?: User;

  @Column()
  userId: string;

  @CreateDateColumn()
  createdAt: Date;

  @Column()
  emote: string;
}

File: libs/gomodels/channel_emote_usage.go
package model

import "time"

type ChannelEmoteUsage struct {
	ID        string    `gorm:"primary_key;column:id;type:TEXT;" json:"id"`
	ChannelID string    `gorm:"column:channelId;type:text"              json:"channelId"`
	UserID    string    `gorm:"column:userId;type:text"                 json:"userId"`
	Emote     string    `gorm:"column:emote;type:text"                 json:"emote"`
	CreatedAt time.Time `gorm:"column:createdAt;type:timestamp"         json:"createdAt"`

	Channel *Channels `gorm:"foreignKey:ID" json:"channel"`
	User    *Users    `gorm:"foreignKey:ID" json:"user"`
}

type ChannelEmoteUsageWithCount struct {
	*ChannelEmoteUsage

	Count int `gorm:"count"`
}

func (c *ChannelEmoteUsage) TableName() string {
	return "channels_emotes_usages"
}
